---
title: "Analysis Report 1: Diversity of Microbial Communities Among Females and Males"
author: "Chelsea Noack"
date: "October 26, 2017"
output: github_document
bibliography: references.bib
csl: bioinformatics.csl
---

# Introduction

Microbial communities are groups of microorganisms inhabiting the same place. These communities span from sizes of large to small. Pairwise interactions between different species within the community predict the microbial community. As bacterial species interact with one another, they in turn affect the health of their host. In a human host, microbes can live in various places on the body, from the gut to the palm of a hand. As humans, we can learn a lot from these communities. 

Skin-associated microbial communities are diverse and personalized. The diversity is due to, first, the skin being a multi-layered organ. Second, each layer has different structures, ranging from arm hair follicles to glands. The different structures operate as sub-habitats for microbes. Third, the skin has various niches a microbe can inhabit. The niches themselves range from pH to moisture levels. The above all attribute to the diversity of skin-associate microbial communities.

The microbial communities found on human skin are also personalized, which raises a lot of questions. By knowing the makeup of one's microbial community, what can we learn about the individual? Further, what can learn about a community whom all share similar microbial communities? What can we learn about the perceived differences? Specifically, can something such as gender be an indicator for microbial community difference? I hypothesize that females and males have different microbial communities. In my analysis, I will attempt to answer this question using the data from a study done by Fierer et al. (2010). In the study,  Fierer et al. took samples from personal computers and the computer owner's palm. By studying the similarities between the samples, Fierer et al. hypothesized that one can identify the computer by its owner using microbial communities [@fierer2010forensic]. Given the data, I will attempt to answer my biological question about the relationship between gender and microbes. 

# Methods

## Sample origin and sequencing

Nine women and men of equivalent age and health (four women, five men) were used for the computer mouse study, all of whom worked in the same building at the University of Colorado. The computer mouse belonging to the individual and the palm of the individual's dominant hand were swabbed. Each computer mouse had been touched by the individual in the last 12 hours. Before sampling, all individuals practiced their regular hygeine routines. Fierer et al. extracted the DNA from the swabs using MO BIO Powersoil isolation kit. Next, they amplified the 16SrRNA genes from every sample using a primer set then running it through PCR. Finally, they sequenced the DNA of each sample by pyrosequencing using a 454 Life Sciences Genome Sequencer FLX Instrument (Roche). If sequences were less than 200 or more than 300 base pairs in length, they were excluded from the study. They compared the swabs from the nine individuals to swabs from 270 other hands from healthy individuals ranging from 18-40 years old. 

## Computational

First, Fierer et al. obtained the sequences from the RDP database which is open for the public. The files were downloaded to a raw data directory. Once the raw data was downloaded in, the package DADA2 [@callahan2016] needed to be installed from bioconductor which serves as the primary package to clean up and analyze our data. In addition, we downloaded phyloseq [@mcmurdie2013]  Using DADA2, we extracted sample names from their original .fastq names and ensured our samples were in order. 

Once all files were downloaded and organized, we did a QC (quality check) report on each individual file.QC reports are helpful to see how viable each sequence is for analysis; if the majority of the sequence is above a 30 score, then we consider it to be sequenced well. Typically, the 3' end will deteriorate, which is normal. Once the files were checked for quality, they were placed in a file in which filtered files would be placed (no files exceeding 225 because quality of bases deteriorate around approximately 200 bp). Finally, the filtered files were trimmed down based on DADA2's recommended 454 parameters, as Fierer et al. used 454 pyrosequencing (Roche). Once the files were trimmed, they are ready to be analyzed.  

# Results

```{r load-libraries, message = FALSE}
# Be sure to install these packages before running this script
# They can be installed either with the intall.packages() function
# or with the 'Packages' pane in RStudio

# load general-use packages
library("dplyr")
library("tidyr")
library("knitr")
library("ggplot2")

# this package allows for the easy inclusion of literature citations in our Rmd
# more info here: https://github.com/crsh/citr
# and here:
# http://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html
library("citr")

# These are the primary packages well use to clean and analyze the data
# this package needs to be installed from bioconductor -- it's not on CRAN
# see info here: https://benjjneb.github.io/dada2/dada-installation.html
library("dada2")

# This to export a fasta of our final denoised sequence variants
library("seqinr")

# To install this you have to install from GitHub
# See more info here: https://github.com/leffj/mctoolsr
# run this -- install.packages("devtools")
# and then this -- devtools::install_github("leffj/mctoolsr")
library("mctoolsr")

# And this to visualize our results
# it also needs to be installed from bioconductor
library("phyloseq")

# Install devtools package
devtools::install_github("ropenscilabs/gramr")

#install word count add in
devtools::install_github("benmarwick/wordcountaddin",
                         type = "source", dependencies = TRUE)
```

```{r extract-sample-and-file-names}
# NOTE: Much of the following follows the DADA2 tutorials available here:
# https://benjjneb.github.io/dada2/tutorial.html
# Accessed October 19, 2017

# set the base path for our input data files
path <- "data/raw_data"

# Sort ensures samples are in order
filenames_forward_reads <- sort(list.files(path, pattern = ".fastq"))

# Extract sample names, assuming filenames have format: SAMPLENAME.fastq
sample_names <- sapply(strsplit(filenames_forward_reads, "\\."), `[`, 1)

# Specify the full path to each of the filenames_forward_reads
filenames_forward_reads <- file.path(path, filenames_forward_reads)
```

```{r check-quality-plots}
# Plots the quality profiles of all twenty samples
plotQualityProfile(filenames_forward_reads[1:20])
```

**Quality check**
We can see from the quality profiles that most reads tend to get pretty bad in quality after around 200 bases. Therefore, we decided to set a maximum acceptable sequence length of 225 bases. For the most part, QC was good. A common, albeit unsurprising, pattern was a degradation as the sequences got closer to the 3' end. However, the degree in which the sequences degraded varied. For example: In sequence ERR1942280, we see that the mean quality of scores for the bases were approximately 36, which is a well-above average score. In other scores such as ERR1942282, the average was 35. While this indicates that both sequences achieved a great sequence score, when looking at the data distribution visually, they look quite different. In reality, most of the sequences provided by Fierer et al. are above 30 and thus very good.

```{r filter-reads}
# Place filtered files in filtered/ subdirectory
# note this will fail if the directory doesn't exist
filter_path <- file.path("output", "filtered")
filtered_reads_path <- file.path(filter_path,
                                 paste0(sample_names,
                                        "_filt.fastq.gz"))

# See ?filterAndTrim for details on the parameters
# See here for adjustments for 454 data:
# https://benjjneb.github.io/dada2/
#     faq.html#can-i-use-dada2-with-my-454-or-ion-torrent-data
filtered_output <- filterAndTrim(fwd = filenames_forward_reads,
                                 filt = filtered_reads_path,
                                 maxLen = 225,
                                 maxN = 0, # discard any seqs with Ns
                                 maxEE = 3, # allow w/ up to 3 expected errors
                                 truncQ = 2, # cut off if quality gets this low
                                 rm.phix = TRUE,
                                 compress = TRUE,
                                 multithread = FALSE)
```

**Trimming**
When the sequences were run through DADA2’s Trim, we found most of the sequences needed to be trimmed. Specifically, 81, 82, 85, 93, and 99 were significantly cut down. While this is important to make note of, I am not as concerned with how these results affect my analysis’ question as to how females and males differ in microbial communities. 

```{r filtered-read-counts-table}
# produce nicely-formatted markdown table of read counts
# before/after trimming
kable(filtered_output,
      col.names = c("Reads In",
                    "Reads Out"))
```

```{r learn-errors}
# this build error models from each of the samples
errors_forward_reads <- learnErrors(filtered_reads_path,
                                    multithread = FALSE)
```

```{r visualize-errors-with-plots}
# quick check to see if error models match data
# (black lines match black points) and are generally decresing left to right
plotErrors(errors_forward_reads,
           nominalQ = TRUE)
```

**Errors**
Not all sequences were the same length, which I expected. Every sample had plenty of errors per sequence for both females and males. They are visualized in the errors with plots. Generally, you want the black line to follow the black points and, as expected, they generally decrease to the left. In terms of errors, outliers were rare, again indicating the sequencing was well done.  

```{r dereplicate-sequences}
# get rid of any duplicated sequences
dereplicated_forward_reads <- derepFastq(filtered_reads_path,
                                         verbose = TRUE)

# Name the derep-class objects by the sample names
names(dereplicated_forward_reads) <- sample_names
```

```{r run-dada}
# parameters adjusted based on recommendations for 454 data here:
# https://benjjneb.github.io/dada2/
#     faq.html#can-i-use-dada2-with-my-454-or-ion-torrent-data
dada_forward_reads <- dada(dereplicated_forward_reads,
                           err = errors_forward_reads,
                           HOMOPOLYMER_GAP_PENALTY = -1, # reduce penalty bc 454
                           BAND_SIZE = 32) # performs local alignments bc indels

# check dada results
dada_forward_reads
```

```{r make-sequence-table}
# produce the 'site by species matrix'
sequence_table <- makeSequenceTable(dada_forward_reads)
```

The output table has `r nrow(sequence_table)` rows (samples) and `r ncol(sequence_table)` columns (sequence variants). Notice how we can embed R code directly in our markdown text. Histogram of sequence lengths: After trimming in DADA2 and denoising, the histogram shows a strong skew to the right with most sequences were around 220-225 bp. The lower range of sequences were below 220 bp. 

```{r histogram-of-sequence-lengths}
# Quick check to look at distribution of trimmed and denoised sequences
hist(nchar(getSequences(sequence_table)),
     main = "Histogram of fingal sequence variant lengths",
     xlab = "Sequence length in bp")
```

```{r remove-chimeras}
# Check for and remove chimeras
sequence_table_nochim <- removeBimeraDenovo(sequence_table,
                                            method = "consensus",
                                            multithread = FALSE,
                                            verbose = TRUE)

# What percent of our reads are non-chimeric?
non_chimeric_reads <- round(sum(sequence_table_nochim) / sum(sequence_table),
                            digits = 4) * 100
```

After removing chimeras, we were left with `r non_chimeric_reads`% of our cleaned reads. Sequence table with no chimeras: 
Using this table because we want to do variant analysis methods because DADA2 does not cluster things. The alternative, OTU, is more error-prone. The table will show all unique sequences in a sample while avoiding clustering. We see 176 sequences in each column with 20 samples in the rows. Only one sequence was shared by 5 samples: 84, 86, 93, 95, 99. Much of this is due to the experimenters parameters and thus while I find that this particular sequence might be significant, I will not focus heavily on its implications to my original question. When looking at the taxa, we see that sequence is match to an unnamed Proteobacteria, which makes sense to be shared by both men and women. When scanning through the rest of the 176 entries, the majority of shared sequences was within the sample. 

```{r table-of-pipeline-read-counts}
# Build a table showing how many sequences remain at each step of the pipeline
get_n <- function(x) sum(getUniques(x)) # make a quick function
track <- cbind(filtered_output, # already has 2 columns
               sapply(dada_forward_reads, get_n),
               rowSums(sequence_table),
               rowSums(sequence_table_nochim))

# add nice meaningful column names
colnames(track) <- c("Input",
                     "Filtered",
                     "Denoised",
                     "Sequence Table",
                     "Non-chimeric")

# set the proper rownames
rownames(track) <- sample_names

# produce nice markdown table of progress through the pipeline
kable(track)
```

```{r assign-taxonomy}
# assigns taxonomy to each sequence variant based on a supplied training set
# made up of known sequences
taxa <- assignTaxonomy(sequence_table_nochim,
                       "data/training/rdp_train_set_16.fa.gz",
                       multithread = FALSE,
                       tryRC = TRUE) # also check with seq reverse compliments

# show the results of the taxonomy assignment
unname(taxa)
```

```{r extract-sequences-to-fasta}
# we want to export the cleaned, trimmed, filtered, denoised sequence variants
# so that we can build a phylogeny - we'll build the phylogeny outside of R
# but we need the fasta file to do so. We keep the names of each sequence as the
# sequence itself (which is rather confusing), because that's how DADA2 labels
# it's columns (e.g. 'species')
# function taken from https://github.com/benjjneb/dada2/issues/88
export_taxa_table_and_seqs <- function(sequence_table_nochim,
                                       file_seqtab,
                                       file_seqs) {
  seqtab_t <- as.data.frame(t(sequence_table_nochim)) # transpose to data frame
  seqs <- row.names(seqtab_t) # extract rownames
  row.names(seqtab_t) <- seqs # set rownames to sequences
  outlist <- list(data_loaded = seqtab_t)
  mctoolsr::export_taxa_table(outlist, file_seqtab) # write out an OTU table
  seqs <- as.list(seqs)
  seqinr::write.fasta(seqs, row.names(seqtab_t), file_seqs) # write out fasta
}

# actually run the function, with the names of the files we want it to create
# and where to put them
export_taxa_table_and_seqs(sequence_table_nochim,
                           "output/sequence_variants_table.txt",
                           "output/sequence_variants_seqs.fa")
```


```{r read-in-metadata-and-create-phyloseq}
# Next we want to read in the metadata file so we can add that in too
# This is not a csv file, so we have to use a slightly different syntax
# here the `sep = "\t"` tells the function that the data are tab-delimited
# and the `stringsAsFactors = FALSE` tells it not to assume that things are
# categorical variables
metadata_in <- read.table(paste0("data/metadata/",
                                 "fierer_forensic_hand_mouse_SraRunTable.txt"),
                          sep = "\t",
                          header = TRUE,
                          stringsAsFactors = FALSE,
                          row.names = 6) # sets sample IDs to row names

# read in the phylogeny, which was created from the fasta exported above
# in Geneious by aligning the sequences with MAFFT and then building a
# Maximum-Likelihood tree with RAxML
tree_in <- read_tree("output/sequence_variants_MAFFT_RAxML.newick")

# Construct phyloseq object (straightforward from dada2 outputs)
phyloseq_obj <- phyloseq(otu_table(sequence_table_nochim,
                                   taxa_are_rows = FALSE), # sample-spp matrix
                         sample_data(metadata_in), # metadata for each sample
                         tax_table(taxa), # taxonomy for each sequence variant
                         phy_tree(tree_in)) # phylogeny from sequence variants
```

```{r meltedtable}
# Compile into a table
melted_obj <- psmelt(phyloseq_obj)
```

```{r subset_phyloseq_obj}
# Make subsetted information without Non-applicable
subsetted_phyloseq_obj <- subset_samples(phyloseq_obj,
                                         sex_s != "Not applicable")
```

**Figure 1**: Inferred phylogeny of classes, with points on tips representing the different classes which occurred. Tree represents maximum likelihood phylogeny inferred using RAxML.

```{r phylogeny-by-phylum}
# phylogeny, yay!
plot_tree(phyloseq_obj,
          color = "Phylum",
          ladderize = TRUE) # this arranges the tree branches from short to long
```

**Figure 2**: Barplot indicating the abundance of certain phylums against the two sexes sampled. 

```{r barplot-by-phylumsexsampleabundance}
library("ggplot2")
p <- plot_bar(subsetted_phyloseq_obj,
             "sex_s",
             fill = "Phylum",
             facet_grid = ~anonymized_name_s)
p + geom_point(aes(x = sex_s, y = Abundance),
               color = "black",
               position = "jitter",
               size = 1)
```

**Figure 3**: Bar plot of abundance of bacteria by classes in females and males.

```{r barplot-by-classsexabundance}
#CLASS PLOT
plot_bar(subsetted_phyloseq_obj,
         x = "Class", fill = "sex_s")
```

**Figure 4**: Bar plot of abundance of bacteria by order in females and males.

```{r barplot-by-ordersexabundance}
# ORDER PLOT
plot_bar(subsetted_phyloseq_obj,
         x = "Order", fill = "sex_s")
```

**Figure 5**: Bar plot of abundance of bacteria by family in females and males.

```{r barplot-by-familysexabundance}
# FAMILY PLOT
plot_bar(subsetted_phyloseq_obj,
         x = "Family", fill = "sex_s")
```

**Figure 6**: Bar plot of abundance of bacteria by genus in females and males.

```{r barplot-by-genussexabundance}
# GENUS PLOT
plot_bar(subsetted_phyloseq_obj,
         x = "Genus", fill = "sex_s")
```


# Discussion

**Figure 1**

The following is a phylogeny of the phylums from the experiment. The rationale behind analyzing it is to get a general picture of the data. We see an over abundance of Firmicutes, Proteobacteria, and Actinobacteria. Thus we can assume prior to any investigation into the difference of the sexes that, most likely, both men and women will have these three phyla. What is in less abundance but still has importance is Cyanobacteria/Chloroplast, Bacteriodetes, and Gemmatimonadetes. 

**Figure 2**

Using a subset of our data to include only females and males, while there was a category for the "Not applicable". Figure 2 confirms our assumption about the overabundance of particular Phylums in both men and women, especially Proteobacteria. However, in contrast to our previous assumption, the majority of the Actinobacteria was found in a single female sample. Not surprisingly, one male sample held the majority of the Cyanobacteria/Chloroplast, which I previously assumed would linger in one of the sexes given its low abundance. 

**Figure 3**

My hypothesis was that females and males differ in their microbial communities and that it would most distinctly differ by class of bacteria. What is first glaring in all the following figures, Figure 3 included, in the massive overabundance of N/A (not applicable) bacteria found on male palms. What concerns me is that this might be indicitive of poor sampling for the males rather than some conspicuous bacteria, but I'm weary to make too broad scale of assumptions. Secondly, we see, as from the previous figure, a high abundance of Actinobacteria in females. Third, our assumption previously made is confirmed that Proteobacteria is spread out fairly evenly amongst the females and males (specifically Alphaproteobacteria and Betaproteobacteria). Lastly, we agains see the male samples dominating in Chloroplast. What IS new is that we see a class called Flavobacteria which is found mainly on females (in Phylum Bacteriodetes). While it is new on the plot, we expected to find a Bacteriodete on mainly one of the sexes hands given the lack of abundance out of all the Phylums. 

**Figure 4**

In Figure 4, we continue to see the glaringly common theme that there is an overabundance of N/A in the male samples. However, we do see Nesseriales in male samples, which comes from class Betaproteobacteria, Proteobacteria. In females, we see Rhizobiales, which comes from class Alphaproteobacteria, Phylyum Proteobacteria. While I have thus far made a distinguishing line between females and males, specifically when it comes to Proteobacteria, Figure 4 also shows us that they share an Order within Betaproteobacteria called Burkholderiales. Lastly, we find that similarly men usually have chloroplast and women have Actinobacteria, the data roughly biased however given we have shown it came from only one male sample and one female sample, which cannot portray the reality of an entire population. 

**Figure 5**

In Figure 5, we predictably see less abundance as the other plots, as this is graphing Family. We see, unsurprisingly, that there is a higher abundance of males in Family Neisseriaceae, Order Neisseriales, Class Betaproteobacteria, Phylum Proteobacteria. In females, also unsurprisingly, we see Family Bartonellaceae, Order Rhizobiales, Class Alphaproteobacteria, Phylum Proteobacteria. Similarly to what was found in Figure 4, they shared a Family within Burkholderiales called Comamonadaceae. 

Some new information popped up from Figure 5. First is that females 

**Figure 6**

**Similarities**
Proteobacteria: Figure 1, 2. Specifically Burkholderiales in Betaproteobacteria. 

**Differences**

Actinobacteria mostly in one female sample, first: Figure 2, Figure 3, Figure 4.

Cyanobacteria/chloroplast mostly in one male sample: Figure 2, Figure 3, Figure 4.

Proteobacteria in both females and males: All figures 
BUT figure 3 shows women going to alpha and men going to beta. 
More interestingly, Figure 4 shows men going to Nesseriales specifically, making my hypothesis wrong that the most indicitave order of Bacteria between females and males (from this study) is Order. 

** Males: Neisseriaceae, Neisseriales, Betaproteobacteria, Proteobacteria **

**Females: Bartonellaceae, Rhizobiales, Alphaproteobacteria, Proteobacteria**

Bacteriodete: Figure 3.

N.A in males: All.

# Errors/Bias

Females and males are identified as such by their sexes, however the study doesn't account for gender. 

# Sources Cited


