---
title: "Analysis Report 1: Diversity of Microbial Communities Among Females and Males"
author: "Chelsea Noack"
date: "October 30, 2017"
output: github_document
bibliography: references.bib
csl: bioinformatics.csl
---

# Introduction

Microbial communities are groups of microorganisms inhabiting the same place. These communities span from sizes of large to small. Pairwise interactions between different species within the community predict the microbial community. As bacterial species interact with one another, they in turn affect the health of their host. In a human host, microbes can live in various places on the body, from the gut to the palm of a hand. As humans, we can learn a lot from these communities. 

Skin-associated microbial communities are diverse and personalized. The diversity is due to, first, the skin being a multi-layered organ. Second, each layer has different structures, ranging from arm hair follicles to glands. The different structures operate as sub-habitats for microbes. Third, the skin has various niches a microbe can inhabit. The niches themselves range from pH to moisture levels. The above all attribute to the diversity of skin-associate microbial communities.

The microbial communities found on human skin are also personalized, which raises a lot of questions. By knowing the makeup of one's microbial community, what can we learn about the individual? Further, what can learn about a community whom all share similar microbial communities? What can we learn about the perceived differences? Specifically, can something such as sex be an indicator for microbial community difference? I hypothesize that females and males have different microbial communities. In my analysis, I will attempt to answer this question using the data from a study done by Fierer et al. (2010). In the study,  Fierer et al. took samples from personal computers and the computer owner's palm. By studying the similarities between the samples, Fierer et al. hypothesized that one can identify the computer by its owner using microbial communities [@fierer2010forensic]. Given the data, I will attempt to answer my biological question about the relationship between gender and microbes. 

# Methods

## Sample origin and sequencing

Nine women and men of equivalent age and health (four women, five men) served as subjects for the computer mouse study. All subjects worked in the same building at the University of Colorado. The computer mouses all belonged to one individual. Every individual touched their computer mouse in the last 12 hours. Before sampling, all individuals practiced their regular hygiene routines. Fierer et al. used an MO BIO Powersoil isolation kit in order to extract DNA from the individuals. The extractions were essentially swabbing the palm.  

 Next, they amplified the 16SrRNA genes from every sample. They did this by using a primer set then running it through PCR. 16SrRNA is a common genetic marker used for sequencing. It's often used because it is present in nearly all bacteria and it is large enough for our informatics purposes (@janda200716s). Finally, Fierer et al. sequenced the DNA of each sample. The method of choice was pyrosequencing using a 454 Life Sciences Genome Sequencer FLX Instrument (Roche). Pyrosequencing is a type of sequencing that follows the "sequencing by synthesis" model. If after sequencing they were less than 200 or more than 300 base pairs in length, Fierer et al. excluded the sequences. 

Finally, they compared the DNA from the nine individuals to DNA swabs from 270 other healthy hands ranging from 18-40 years old. 

## Computational

First, Fierer et al. obtained the sequences from the RDP database, which is open for the public. We downloaded the files to a raw data directory. 

A package called  DADA2 [@callahan2016] needed to be installed. The package came from a bioconductor which serves as the primary package to clean up and analyze our data. We also downloaded another package called phyloseq [@mcmurdie2013]. Once DADA2 and phyloseq downloaded, we used DADA2 to extract sample names from their original .fastq names. We also ensured our samples were in order. When all files were downloaded and organized, we did a QC (quality check) report on each individual file. 

QC reports help us see how viable each sequence is for analysis. If the majority of the sequence is above a 30 score, then we consider it to be sequenced well. Typically, the 3' end will deteriorate, which is normal. After the quality check, all filtered files (no files exceeding 225 because quality of bases deteriorate around approximately 200 bp) were placed in one file. Finally, the filtered files were trimmed down using DADA2's recommended 454 parameters, as Fierer et al. used 454 pyrosequencing (Roche). Once the files were trimmed, they are ready to be analyzed. 

# Results

```{r load-libraries, echo=FALSE, message = FALSE}
# Be sure to install these packages before running this script
# They can be installed either with the intall.packages() function
# or with the 'Packages' pane in RStudio

# load general-use packages
library("dplyr")
library("tidyr")
library("knitr")
library("ggplot2")

# this package allows for the easy inclusion of literature citations in our Rmd
# more info here: https://github.com/crsh/citr
# and here:
# http://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html
library("citr")

# These are the primary packages well use to clean and analyze the data
# this package needs to be installed from bioconductor -- it's not on CRAN
# see info here: https://benjjneb.github.io/dada2/dada-installation.html
library("dada2")

# This to export a fasta of our final denoised sequence variants
library("seqinr")

# To install this you have to install from GitHub
# See more info here: https://github.com/leffj/mctoolsr
# run this -- install.packages("devtools")
# and then this -- devtools::install_github("leffj/mctoolsr")
library("mctoolsr")

# And this to visualize our results
# it also needs to be installed from bioconductor
library("phyloseq")

# Install devtools package
devtools::install_github("ropenscilabs/gramr")

#install word count add in
devtools::install_github("benmarwick/wordcountaddin",
                         type = "source", dependencies = TRUE)
```

```{r extract-sample-and-file-names, echo=FALSE}
# NOTE: Much of the following follows the DADA2 tutorials available here:
# https://benjjneb.github.io/dada2/tutorial.html
# Accessed October 19, 2017

# set the base path for our input data files
path <- "data/raw_data"

# Sort ensures samples are in order
filenames_forward_reads <- sort(list.files(path, pattern = ".fastq"))

# Extract sample names, assuming filenames have format: SAMPLENAME.fastq
sample_names <- sapply(strsplit(filenames_forward_reads, "\\."), `[`, 1)

# Specify the full path to each of the filenames_forward_reads
filenames_forward_reads <- file.path(path, filenames_forward_reads)
```

```{r check-quality-plots}
# Plots the quality profiles of all twenty samples
plotQualityProfile(filenames_forward_reads[1:20])
```

**Quality check**: The quality profiles show that most reads tend to decline in quality after 200 bases. Given the deterioration, we decided to set a maximum acceptable sequence length of 225 bases. Overall however, QC was good. A common, albeit unsurprising, pattern was a degradation as the sequences got closer to the 3' end. Yet the degree in which the sequences degraded varied. 

For example: In sequence ERR1942280, we see that the mean quality of scores for the bases were approximately 36, which is a well-above average score. In other scores such as ERR1942282, the average was 35. While both sequences achieved a great sequence score, visually they look quite different. In reality, most of the sequences provided by Fierer et al. are above 30 and thus very good.

```{r filter-reads, echo=FALSE}
# Place filtered files in filtered/ subdirectory
# note this will fail if the directory doesn't exist
filter_path <- file.path("output", "filtered")
filtered_reads_path <- file.path(filter_path,
                                 paste0(sample_names,
                                        "_filt.fastq.gz"))

# See ?filterAndTrim for details on the parameters
# See here for adjustments for 454 data:
# https://benjjneb.github.io/dada2/
#     faq.html#can-i-use-dada2-with-my-454-or-ion-torrent-data
filtered_output <- filterAndTrim(fwd = filenames_forward_reads,
                                 filt = filtered_reads_path,
                                 maxLen = 225,
                                 maxN = 0, # discard any seqs with Ns
                                 maxEE = 3, # allow w/ up to 3 expected errors
                                 truncQ = 2, # cut off if quality gets this low
                                 rm.phix = TRUE,
                                 compress = TRUE,
                                 multithread = FALSE)
```

```{r filtered-read-counts-table, echo=FALSE}
# produce nicely-formatted markdown table of read counts
# before/after trimming
kable(filtered_output,
      col.names = c("Reads In",
                    "Reads Out"))
```

```{r learn-errors, echo=FALSE}
# this build error models from each of the samples
errors_forward_reads <- learnErrors(filtered_reads_path,
                                    multithread = FALSE)
```

```{r visualize-errors-with-plots}
# quick check to see if error models match data
# (black lines match black points) and are generally decresing left to right
plotErrors(errors_forward_reads,
           nominalQ = TRUE)
```

**Errors**
Not all sequences were the same length, which I expected. Every sample had plenty of errors per sequence for both females and males. Generally, you want the black line to follow the black points and, as expected, they generally decrease to the left. In terms of errors, outliers were rare, again indicating the sequencing was well done.  

```{r dereplicate-sequences, echo=FALSE}
# get rid of any duplicated sequences
dereplicated_forward_reads <- derepFastq(filtered_reads_path,
                                         verbose = TRUE)

# Name the derep-class objects by the sample names
names(dereplicated_forward_reads) <- sample_names
```

```{r run-dada, echo=FALSE}
# parameters adjusted based on recommendations for 454 data here:
# https://benjjneb.github.io/dada2/
#     faq.html#can-i-use-dada2-with-my-454-or-ion-torrent-data
dada_forward_reads <- dada(dereplicated_forward_reads,
                           err = errors_forward_reads,
                           HOMOPOLYMER_GAP_PENALTY = -1, # reduce penalty bc 454
                           BAND_SIZE = 32) # performs local alignments bc indels

# check dada results
dada_forward_reads
```

```{r make-sequence-table, echo=FALSE}
# produce the 'site by species matrix'
sequence_table <- makeSequenceTable(dada_forward_reads)
```

```{r histogram-of-sequence-lengths}
# Quick check to look at distribution of trimmed and denoised sequences
hist(nchar(getSequences(sequence_table)),
     main = "Histogram of fingal sequence variant lengths",
     xlab = "Sequence length in bp")
```

**Histogram of sequence lengths**: After trimming in DADA2 and denoising, the histogram shows a strong skew to the right with most sequences were around 220-225 bp. The lower range of sequences were below 220 bp. 

```{r remove-chimeras}
# Check for and remove chimeras
sequence_table_nochim <- removeBimeraDenovo(sequence_table,
                                            method = "consensus",
                                            multithread = FALSE,
                                            verbose = TRUE)

# What percent of our reads are non-chimeric?
non_chimeric_reads <- round(sum(sequence_table_nochim) / sum(sequence_table),
                            digits = 4) * 100
```

**Chimera Removal**: After removing chimeras, we have `r non_chimeric_reads`% of our cleaned reads. We use a table because we want to do variant analysis methods and DADA2 does not cluster things. The alternative, OTU, is more error-prone. The table will show all unique sequences in a sample while avoiding clustering. 

We see 176 sequences in each column with 20 samples in the rows.  5 samples (84, 86, 93, 95, 99) shared one sequence. This is due to the experimenters parameters. When looking at the taxa, we see that sequence is match to an unnamed Proteobacteria, which makes sense as later on we see that the majority of females and males have a type of Proteobacteria.

```{r table-of-pipeline-read-counts, echo=FALSE}
# Build a table showing how many sequences remain at each step of the pipeline
get_n <- function(x) sum(getUniques(x)) # make a quick function
track <- cbind(filtered_output, # already has 2 columns
               sapply(dada_forward_reads, get_n),
               rowSums(sequence_table),
               rowSums(sequence_table_nochim))

# add nice meaningful column names
colnames(track) <- c("Input",
                     "Filtered",
                     "Denoised",
                     "Sequence Table",
                     "Non-chimeric")

# set the proper rownames
rownames(track) <- sample_names

# produce nice markdown table of progress through the pipeline
kable(track)
```

```{r assign-taxonomy, echo=FALSE}
# assigns taxonomy to each sequence variant based on a supplied training set
# made up of known sequences
taxa <- assignTaxonomy(sequence_table_nochim,
                       "data/training/rdp_train_set_16.fa.gz",
                       multithread = FALSE,
                       tryRC = TRUE) # also check with seq reverse compliments

# show the results of the taxonomy assignment
unname(taxa)
```

```{r extract-sequences-to-fasta, echo=FALSE}
# we want to export the cleaned, trimmed, filtered, denoised sequence variants
# so that we can build a phylogeny - we'll build the phylogeny outside of R
# but we need the fasta file to do so. We keep the names of each sequence as the
# sequence itself (which is rather confusing), because that's how DADA2 labels
# it's columns (e.g. 'species')
# function taken from https://github.com/benjjneb/dada2/issues/88
export_taxa_table_and_seqs <- function(sequence_table_nochim,
                                       file_seqtab,
                                       file_seqs) {
  seqtab_t <- as.data.frame(t(sequence_table_nochim)) # transpose to data frame
  seqs <- row.names(seqtab_t) # extract rownames
  row.names(seqtab_t) <- seqs # set rownames to sequences
  outlist <- list(data_loaded = seqtab_t)
  mctoolsr::export_taxa_table(outlist, file_seqtab) # write out an OTU table
  seqs <- as.list(seqs)
  seqinr::write.fasta(seqs, row.names(seqtab_t), file_seqs) # write out fasta
}

# actually run the function, with the names of the files we want it to create
# and where to put them
export_taxa_table_and_seqs(sequence_table_nochim,
                           "output/sequence_variants_table.txt",
                           "output/sequence_variants_seqs.fa")
```


```{r read-in-metadata-and-create-phyloseq, echo=FALSE}
# Next we want to read in the metadata file so we can add that in too
# This is not a csv file, so we have to use a slightly different syntax
# here the `sep = "\t"` tells the function that the data are tab-delimited
# and the `stringsAsFactors = FALSE` tells it not to assume that things are
# categorical variables
metadata_in <- read.table(paste0("data/metadata/",
                                 "fierer_forensic_hand_mouse_SraRunTable.txt"),
                          sep = "\t",
                          header = TRUE,
                          stringsAsFactors = FALSE,
                          row.names = 6) # sets sample IDs to row names

# read in the phylogeny, which was created from the fasta exported above
# in Geneious by aligning the sequences with MAFFT and then building a
# Maximum-Likelihood tree with RAxML
tree_in <- read_tree("output/sequence_variants_MAFFT_RAxML.newick")

# Construct phyloseq object (straightforward from dada2 outputs)
phyloseq_obj <- phyloseq(otu_table(sequence_table_nochim,
                                   taxa_are_rows = FALSE), # sample-spp matrix
                         sample_data(metadata_in), # metadata for each sample
                         tax_table(taxa), # taxonomy for each sequence variant
                         phy_tree(tree_in)) # phylogeny from sequence variants
```

```{r meltedtable, echo=FALSE}
# Compile into a table
melted_obj <- psmelt(phyloseq_obj)
```

```{r subset_phyloseq_obj, echo=FALSE}
# Make subsetted information without Non-applicable
subsetted_phyloseq_obj <- subset_samples(phyloseq_obj,
                                         sex_s != "Not applicable")
```

**Figure 1**: Inferred phylogeny of classes, with points on tips representing the different phylas. Tree represents maximum likelihood phylogeny inferred using RAxML. The rationale behind analyzing it is to get a general picture of the data.

```{r phylogeny-by-phylum}
# Phylogeny code
plot_tree(phyloseq_obj,
          color = "Phylum",
          ladderize = TRUE) # this arranges the tree branches from short to long
```

**Figure 2**: Barplot indicating the abundance of certain phylums against the two sexes sampled, excluding the not applicable category. 

```{r barplot-by-phylumsexsampleabundance}
# Bar plot code with a grid, can analyze sample by sample!
library("ggplot2")
p <- plot_bar(subsetted_phyloseq_obj,
             "sex_s",
             fill = "Phylum",
             facet_grid = ~anonymized_name_s)
p + geom_point(aes(x = sex_s, y = Abundance),
               color = "black",
               position = "jitter",
               size = 1)
```

**Figure 3**: Bar plot of abundance of bacteria by classes in females and males, excluding the not applicable category.

```{r barplot-by-classsexabundance}
# Basic bar plot code showing class 
plot_bar(subsetted_phyloseq_obj,
         x = "Class", fill = "sex_s")
```

**Figure 4**: Bar plot of abundance of bacteria by order in females and males, excluding the not applicable category.

```{r barplot-by-ordersexabundance}
# Basic bar plot code showing order
plot_bar(subsetted_phyloseq_obj,
         x = "Order", fill = "sex_s")
```

**Figure 5**: Bar plot of abundance of bacteria by family in females and males, excluding the not applicable category.

```{r barplot-by-familysexabundance}
# Basic bar plot code showing family
plot_bar(subsetted_phyloseq_obj,
         x = "Family", fill = "sex_s")
```

**Figure 6**: Bar plot of abundance of bacteria by genus in females and males, excluding the not applicable category.

```{r barplot-by-genussexabundance}
# Basic bar plot code showing genus
plot_bar(subsetted_phyloseq_obj,
         x = "Genus", fill = "sex_s")
```

# Discussion

The skin-associated microbial communities differ on females. From the Fierer et al. data, we see that females and males both have an overabundance of Proteobacteria in their DNA samples. Most interestingly, females have more Alphaproteobacteria while males have more Betaproteobacteria. Thus, the sexes appear to diverge at the class level. However, there were similarities between the sexes which deserve clarification. 

# Similarities

Females and males have some similarity when it comes to skin-associated micro biomes. In the data, we see females and males both have an overabundance of Phylum Proteobacteria (Figures 1 & 2). Within Proteobacteria, they share Class Betaproteobacteria (Figure 3). They even share Order Burkholderiales (Figure 4), Family Comamonadaceae (Figure 5), Genus Acidovorax (Figure 6). 

The Acidovorax genus is vast with many species. 

We cannot tell from the given samples the species which appears in both female and male samples. The alternative reasoning is that females and males have different species of the same genus Acidovorax. Not much information about Acidovorax and its relationship with humans is readily available. This means more research needs to occur in order to understand why females and males share the genus. More importantly, it's important to note that this was a small experiment which is prone to sampling error. So, we must take the Acidovorax similarity with a grain of salt. 

One final, albeit small, similarity is the overall an overabundance of Phylum Firmicutes, Proteobacteria, and Actinobacteria (Figure 1, Figure 2) between females and males. The same result was also found by another study done by Fierer et al. in 2008 (@fierer2008influence). The similarity indicates that, most likely, both men and women will have these three phyla.

# Differences

Females and males differ on a small and large scale. On a small scale, females show dominance in class Flavobacteria.  On a large scale, females show dominance in phylum Actinobacteria and males show dominance in Chloroplast. However, the biggest difference occurs between females and males in Proteobacteria. Females were more prone to be alpha-proteobacteria. Males are more prone to be beta-proteobacteria. These three differences are equally important and I will spend time analyzing how they all affect my original hypothesis.

**Flavobacteria**

Cyanobacteria/Chloroplast, Bacteriodetes, and Gemmatimonadetes are less common in the two sexes. Class Flavobacteria within phyla Bacteriodetes, however, are female samples (Figure 3). 

The reason Flavobacterium is interesting is because the bacteria is usually a "causative agent of bacterial cold water disease and rainbow trout fry syndrome" [@nematollahi2003flavobacterium]. How it ended up in female samples is certainly thought-provoking. The bacteria resides mainly in water systems. A previous study claims Flavobacteria occurs "with α-proteobacteria and γ-proteobacteria, [all] dominate[s] the bacterial populations in the Southern Ocean”. They go on to mention that the relationship occurs in other oceanic areas as well [@abell2005ecological].  This information is relevant to our analysis because we saw alpha-proteobacteria and beta-proteobacteria occur in the samples as well. 

The occurrence of Flavobacteria is not an isolated event; it is probably due to a mixture of events. For example, washing hands requires a water source. This could easily be the reasoning why this occurred more in females. Or, potentially, the presence of Flavobacteria might be do to its relationship to another bacteria in our sample. Such reasoning would require further analysis. However, in reference to our hypothesis, we do see this class of Bacteria more in females, which supports that females and males differ in skin-associated micro biomes. 

**Actinobacteria & Chloroplast**

Actinobacteria is more prominent in females (Figures 2, 3). Actinobacteria is a gram-positive bacteria with a lot of GC content in their DNA [@ventura2007genomics]. When a phylum is as big and biodiverse as Actinobacteria, it is hard to gesture at what or why is driving only females to have it. However, the order of bacteria, Actinomycetales, is where is got interesting. Many species f Actinomycetales occur in the soil. Perhaps the female students had been near the soil recently, as this was Colorado, and even after washing their hands couldn't get it off. Opposingly, Actinomycetales is an important pathogen and source of antibiotics, which could mean something entirely different. Because the phylum occurred in multiple female samples, this would be worth investigating. For my purposes, it merely reinforces that there is a difference between female and male micro biomes. 

Chloroplast came from a single male sample, unlike Actinobacteria (Figures 2, 3, & 4). In regard to my hypothesis and analysis, I do not find this data to be supportive. When a Phylum occurs only in one sample, I argue it cannot speak for the sex and thus cannot help support that females and males differ. 

**Proteobacteria in Females and Males**

Perhaps the biggest difference between females and males was within Proteobacteria. Proteobacteria appears in both females and males throughout every figure. Females are more likely alpha-proteobacteria and males likely have beta-proteobacteria (Figure 3). 

Alpha and beta-proteobacteria differ in shape and physiology. Yet belong to the same phylum of gram-negative, purple bacteria [@stackebrandt1988proteobacteria]. Why would females have alpha-proteobacteria? And on the flip side, what drives males to have beta-proteobacteria? Answers to these questions do not belong here, as our data is not big enough. Given that the samples live and work near one another, the results might be due to geography less so much as life history or genetics. But when we look deeper within the sample (Figures 4, 5, &6), we see our questions narrow.

Within beta-proteobacteria for males, we see Family Neisseriaceae, Order Neisseriales, Class Beta-proteobacteria, Phylum Proteobacteria (Figures 3, 4, &5). The genus Neisseria is usually a primary pathogen in humans responsible for two household names, namely gonorrhea and meningitis [@stern1986opacity] [@tettelin2000complete]. Ethically, findings like these are challenging. Once common words of afflictions such as these arise, judgements come up. Immediately, the findings cause a desire for informaticians to create a bias of how we paint the picture of the sample. In reality, bacteria is fickle because it can pertain to many things given its environment/host. Yes, there is a possibility some male samples have meningitis or, worse, gonorrhea. It is not within our power to make overarching assumptions because we have a lot of data and its tools, which we cannot trust anyway. In terms of our hypothesis, this is one of the biggest differences between males and females because it narrows down to the genus of the phylum and it occurs throughout many samples.  

In females, we see Genus Bartonella, Family Bartonellaceae, Order Rhizobiales, Class Alphaproteobacteria, Phylum Proteobacteria. Unlike the primary pathogen behavior of Neisseria, Bartonella is an opportunistic pathogen. An opportunistic pathogen means it comes from another pathogen. While it can live on healthy humans, it transfers from sand flies, ticks, and the like. Glaringly, seven out of nineteen species of Bartonella connect to human disease [@zeaiter2002genetic]. This information requires further research between females and Bartonella, specifically within the samples from Colorado. Given I cannot do that, I can hypothesize carefully, traipsing the same ethical line I drew with the males. More importantly, this reinforces my hypothesis that females and males differ in their skin-associated micro biomes. 

# What It Means

The data confirms my general hypothesis, that females and males differ in skin-associated micro biomes. But what does it mean?

The difference can mean that females and males differ in their hygiene practice. Before the study, the subjects went about their normal practices when it comes to hand washing. Thus, the results can be telling us less so much about sex on the genetic basis, but more of a behavior difference. The change in a bacterial community composition since the last time an individual washes their hands is significant [@fierer2008influence]. In addition, while females and males might have distinct bacterial communities, they become more apparent since hand washing [@fierer2008influence]. It is not far-fetched to say that hand washing plays a vital part when the palm was the area of DNA extraction. 

Hand washing is not the only reason females and males may differ in microbial communities. It might be due to difference in pH, sweat production, cosmetics, moisturizers, thickness of skin, or even hormone production [@fierer2008influence]. Bacteria, as mentioned before, is fickle because while it provides beautiful data and figures, it's not easily interpretable. 

And not only is bacteria difficult to make assumptions about, but bacteria found on the skin is dynamic. Skin is in contact with many things, especially on the hand. If we want to want to understand skin-associated microbial communities on a deeper level, the data requires further analysis. Clearly, hand washing has a quick effect on an individuals bacterial make up. As we saw, some samples were jarring, such as the males having Neisseria. These results speak to the known importance of proper personal hygeine to prevent us from primary pathogens such as Neisseria. However, we also must be careful with our findings from such research, as the data can only give data, but we as the informaticians map the story. 

Overall, the complexity and interest of bacteria found on the skin is clear. We do see a distinction between females and males. But, informaticians only shape the story to fit what the data provides in the most unbiased way. Thus, when it comes to microbial communities and their relationship to humans, we require more analyses from different perspectives in order to absorb the influx of data now sequenced. 

# Errors/Bias

Like all analysis, this one is apt to error and bias. Firstly, sex is the label for females and males however the study doesn't account for gender. It makes an assumption. While gender is the human confluence, these days its appropriate to say that it can alter behavior. Thus, it can alter how or what micro biomes will associate on the skin. Second, a lot of males are non applicable. This means something went south during sequencing. We can also see this in the error output. Arguably, this troubles much of the data. For our purposes, it didn't encroach too much on the hypothesis. Lastly, the sample itself is small and requires further research. Isolated computer mouses are not a good reflection of the world as a whole and thus the study cannot speak to all females and males. 

# Sources Cited


